### PR TITLE
Adding proper meth recipe

### PR DESCRIPTION
**Pull request by Nabo92**
#### Summary
Content "New chemical ingredients and adjust meth recipe"
Added chemical recipe for making high-grade meth, the high-grade meth the two chemical items needed that weren't in game. Added these and the other chemicals needed for the recipe to the loot pool of the chemistry, mobile and basement lab. Also added acetic anhydride because I intend on making a recipe for the synthesis of heroin. Also, removed old items from meth lab truck and basement.
The new high grade meth has slightly better stats than the low grade one. The downside is that it takes a lot of chemicals, chemistry skill and it takes 6 hours to complete (should maybe be a bit longer, but I guess a skilled chemist could do it in this time).

#### Purpose of change

I was always bothered by the fact that the game only had low-grade methamphetamine, and since this is a pretty easy synthesis IRL, I added the recipe for it.

#### Describe the solution

The current recipe for making the low quality meth is absolutely unrealistic, I could maybe add another recipe with the same ingredients and lower skill, but I guess it wouldn't make sense for the player to make it, so I'd just remove it from the game entirely. I could easily do it in another PR (along with the heroin recipe) if this one is well recieved (it's mt first time, be gentle with me).

#### Describe alternatives you've considered

Just removing the recipe for low quality meth, but that would be lazy.

#### Testing

Got into a debug world, spawned the new items, tested their weights, spawned a meth truck (loot pool was indeed altered), and made the new meth. Seems to be working alright. Didn't go far enough as to get into a lab or meth basement and check if the items were there, but I guess it worked, since they spawned on the mobile meth lab.

#### Additional context
I didn't add the solvents necessary for it in the mobile and basement labs, because that'd be too easy for the player, the way it's implemented means that the player HAS to explore the lab, which I guess gives more reason for a player to go there.
Some additional details concerning realism: the batch savings time is really high because that's how it usually works in real life, making a chemical is a lot of work and it doesn't matter much if you make 1g or 10g, the amount of time is roughly the same.
Hope this is my first of many PRs of adding more chemical stuff to the game, still intend on implementing the heroin and kevlar synthesis and the obtaining of red phosphorous from matches.
Also, if someone has more ideas for chemical recipes that could be useful to a cataclysm survivor, please do comment and I'll see if I can get it done.

### CODE REVIEW

**Comment by Maleclypse** [comment]
I can't judge your chemistry but the json looked good when I looked through.  I'm going to edit your PR Summary for you because it's complicated the first time and it'll make more sense once you see what it should look like.

**Comment by Maleclypse** [comment]
https://dev.narc.ro/cataclysm/format.html can be used to fix the json style errors.

**Comment by Nabo92** [comment]
Yeah, it looked fine on notepad++, but something happened when transferring it back to github, will use it next time though

**Comment by Nabo92** [comment]
I've passed them through the linter, but I have no idea how to edit this PR to reflect this, I have almost 0 coding experience

File: data/json/items/chemicals_and_resources.json [hunk 1]
```diff
@@ -1687,5 +1687,62 @@
...
+    "color": "red",
+    "price": 50,
+    "price_postapoc": 2,
+    "count": 100
+  },
+  {
+    "id": "acetic_anhydride",
+    "type": "COMESTIBLE",
+    "category": "chems",
+    "name": { "str": "acetic anhydride" },
```
Comment (Qrox) [inline data/json/items/chemicals_and_resources.json:1712]:
```suggestion
    "name": { "str_sp": "acetic anhydride" },
```

File: data/json/items/comestibles/med.json [hunk 2]
```diff
@@ -2025,5 +2025,30 @@
     "addiction_type": "cocaine",
     "flags": [ "NO_INGEST", "WATER_DISSOLVE" ],
     "use_action": { "type": "heal", "bleed": 20, "move_cost": 100, "effects": [ { "id": "pkill1", "duration": 720 } ] }
+  },
+  {
+    "id": "pure_meth",
+    "type": "COMESTIBLE",
+    "comestible_type": "MED",
+    "name": { "str_sp": "high-grade methamphetamine" },
+    "description": "A profoundly addictive and powerful stimulant.  While extremely effective at enhancing alertness, it is hazardous to health and the risk of an adverse reaction is great. This one could have only been made by a great chemist",
```
Comment (chaohedgehogs) [inline data/json/items/comestibles/med.json:2034]:
```suggestion
    "description": "A profoundly addictive and powerful stimulant.  While extremely effective at enhancing alertness, it is hazardous to health and the risk of an adverse reaction is great. This one could have only been made by a great chemist.",
```

File: data/json/items/chemicals_and_resources.json (hunk 1 above)
Comment (Nabo92) [reply data/json/items/chemicals_and_resources.json:1712]: I'm accepting suggestion, but I don't know what's the difference between str and str_sp

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/realistic-chemistry-recipes/26802/8

File: data/json/items/chemicals_and_resources.json (hunk 1 above)
Comment (Nabo92) [reply data/json/items/chemicals_and_resources.json]: Im accepting the changes but I don't know the difference between the two strings

Comment (Qrox) [reply data/json/items/chemicals_and_resources.json]: `str_sp` makes the string to have the same plural form as the singular one.

**Comment by Xaleth** [comment]
It's about time that we reach something that is 99.1% pure

**Comment by Nabo92** [comment]
I'd appreciate if someone could help me figure out what went wrong with the workflow runs, because for a noob-ish coder as me, that seems like it should've passed.

**Pull request merged**